modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java

```diff
@@ -427,6 +429,16 @@ public Redirect getRedirect(
 
 				actualURL = HttpComponentsUtil.setParameter(
 					actualURL, "doAsUserId", encDoAsUserId);
+
+				if (params.containsKey("doAsUserId")) {
+					Map<String, String[]> modifiedParams = new HashMap<>();
+
+					MapUtil.copy(params, modifiedParams);
+
+					modifiedParams.remove("doAsUserId");
+
+					params = modifiedParams;
+				}
```

Comment:
Can you simplify the code to:
```
params = new HashMap<>(params);

params.remove("doAsUserId");
```
I don't think checking for the key presence is necessary here, and the passing the original map to the constructor will create a copy of the original.

Reply:
You're right, initially I included that checking for the key presence because I worried that copying the Map might be an expensive operation, so I wanted to avoid it when possible. However, thinking on this further, this didn't make sense, because there should not ever be very many params in a given URL (certainly not enough to make Map copy take a while).

I've edited the code as you suggested, can you please check it when you get a chance? Thanks!